packages/flutter/lib/src/material/text_selection_theme.dart

```diff
@@ -120,25 +120,37 @@ class TextSelectionThemeData with Diagnosticable {
 /// color with light blue selection handles to the child text field.
 ///
 /// ```dart
-/// const TextSelectionTheme(
-///   data: TextSelectionThemeData(
+/// TextSelectionTheme(
+///   data: const TextSelectionThemeData(
 ///     cursorColor: Colors.blue,
 ///     selectionHandleColor: Colors.lightBlue,
 ///   ),
-///   child: TextField(),
+///   child: const TextField(),
 /// )
 /// ```
 /// {@end-tool}
+///
+/// This widget also creates a [DefaultSelectionStyle] for its subtree with
+/// [data].
 class TextSelectionTheme extends InheritedTheme {
   /// Creates a text selection theme widget that specifies the text
   /// selection properties for all widgets below it in the widget tree.
   ///
   /// The data argument must not be null.
-  const TextSelectionTheme({
+  TextSelectionTheme({
     Key? key,
     required this.data,
     required Widget child,
-  }) : assert(data != null), super(key: key, child: child);
+  }) : assert(data != null),
+      super(key: key, child: _wrapsWithDefaultSelectionStyle(data, child));
+
+  static Widget _wrapsWithDefaultSelectionStyle(TextSelectionThemeData data, Widget child) {
```

Comment:
how would that help with making TextSelectionTheme const? the use the const widget in the initializer, the input also needs to be const value.

so I can't do
```dart
super(key: key, child: const MyWidget(data: data, child: child));
```
because the data and child is runtime value.

Reply:
Even if it's not possible to use a const constructor for a ` _WithDefaultSelectionStyle()` widget, it seems clearer and more in-line with Flutter conventions than using a function that has the same effect.